portfolio/src/test/java/com/google/sps/TimeRangeGroupArrayListTest.java

```diff
@@ -0,0 +1,166 @@
+package com.google.sps;
+
+import com.google.sps.data.TimeRange;
+import com.google.sps.data.TimeRangeGroupArrayList;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TimeRangeGroupArrayListTest {
+
+  /** Tests for the method that checks if a time range exists in the group. */
+  @Test
+  public void testHasTimeRange() {
+    // Time Ranges: |--A----|   |---B---|
+    // To check:      |-C-|   |---D---|   |--E--|
+    // C exists; D and E both don't exist in the group
+    Instant timeRangeOneStart = Instant.now();
+    Instant timeRangeOneEnd = timeRangeOneStart.plusSeconds(1000);
+    Instant timeRangeTwoStart = timeRangeOneEnd.plusSeconds(500);
+    Instant timeRangeTwoEnd = timeRangeTwoStart.plusSeconds(1000);
+
+    TimeRange timeRangeOne = TimeRange.fromStartEnd(timeRangeOneStart, timeRangeOneEnd);
+    TimeRange timeRangeTwo = TimeRange.fromStartEnd(timeRangeTwoStart, timeRangeTwoEnd);
+
+    List<TimeRange> timeRanges = Arrays.asList(timeRangeOne, timeRangeTwo);
+    TimeRangeGroupArrayList timeRangeGroup =
+        new TimeRangeGroupArrayList(
+            timeRanges, TimeRange.sortByTimeRangeStartTimeAscending, /* ascending= */ true);
```

Comment:
As I mentioned, I think sorting should be done internally by TimeRangeGroup rather than having the caller specify the sorting approach like this. For example, hash set uses a tree structure internally to keep track of the entries, but the caller doesn't need to specify how to organize the tree. So I think you shouldn't have to pass the comparator and boolean in the constructor.